documentation/Getting started with TypeSpec specifications.md

```diff
@@ -0,0 +1,29 @@
+## What is an TypeSpec Specification?
```

Comment:
Try to avoid spaces in the name of the file as it makes linking to it a little less friendly.

Reply:
Is the idea to replace the content in https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md?

Reply:
No, this is aimed as a single doc entry point. As you see below, it fan out to individual docs so that the docs pages are not too big and overwhelming.